tiling_segmentation.py

```diff
@@ -184,55 +186,169 @@ def make_dataset_file_name(exp_name: str, min_annot: Number, dataset: str, attr_
         @param out_label_dir: optional, path to output patched labels directory
         @return: written patches to output directories as .tif for imagery and .geojson for label.
-        TODO: replace solaris with GDAL and ogr2ogr, generate individual patch bounds with torchgeo's GridGeoSampler.
-        Implies the implementation of RasterDataset & VectorDataset
-        https://gis.stackexchange.com/questions/14712/splitting-raster-into-smaller-chunks-using-gdal
-        https://gdal.org/programs/ogr2ogr.html#ogr2ogr
-        https://gis.stackexchange.com/questions/303979/clip-all-layers-of-a-geopackage-gpkg-in-one-step
+
+        https://torchgeo.readthedocs.io/en/stable/tutorials/custom_raster_dataset.html
+        https://torchgeo.readthedocs.io/en/stable/api/samplers.html
+        https://gdal.org/api/python/osgeo.gdal.html
+        https://gdal.org/api/python/osgeo.ogr.html
         """
-        if not aoi.raster:  # in case of multiprocessing
+
+        if not aoi.raster:
             aoi.raster = rasterio.open(aoi.raster_multiband)
 
-        raster_tiler = tile.raster_tile.RasterTiler(dest_dir=out_img_dir,
-                                                    src_tile_size=(self.dest_patch_size, self.dest_patch_size),
-                                                    alpha=False,
-                                                    verbose=True)
-        # FIXME: workaround for successful writing of patches. 'VRT' driver cannot be assigned to written files. Error:
-        # rasterio.errors.RasterioIOError: Read or write failed. Writing through VRTSourcedRasterBand is not supported.
-        aoi.raster.driver = "GTiff" if aoi.raster.driver == 'VRT' else aoi.raster.driver
-        raster_bounds_crs = raster_tiler.tile(aoi.raster, dest_fname_base=aoi.raster_name.stem)
-        logging.debug(f'Raster bounds crs: {raster_bounds_crs}\n'
-                      f'')
-
-        vec_tler_patch_paths = [None] * len(raster_tiler.tile_paths)
+        # Create TorchGeo-based custom VRTDataset dataset:
+        vrt_dataset = VRTDataset(aoi.raster)
+
         if not self.for_inference:
-            dest_crs = raster_tiler.dest_crs if raster_tiler.dest_crs.is_epsg_code else None
-            vec_tler = tile.vector_tile.VectorTiler(dest_dir=out_label_dir,
-                                                    dest_crs=dest_crs,
-                                                    verbose=True,
-                                                    super_verbose=self.debug)
-            vec_tler.tile(src=str(aoi.label),
-                          tile_bounds=raster_tiler.tile_bounds,
-                          dest_fname_base=aoi.raster_name.stem,
-                          split_multi_geoms=False)
-            vec_tler_patch_paths = vec_tler.tile_paths
+            vec_dataset = GDLVectorDataset(aoi.label)
+            # Combine raster and vector datasets using AND operator (sampling only from the intersection area).
```

Comment:
Is this the same behaviour as previous script? See: https://github.com/NRCan/geo-deep-learning/pull/386/files#r1029641884

Reply:
See the previous comment.